### PR TITLE
Better handle CurseForge manifest.json

### PR DESCRIPTION
**Pull request by hannah98**
Pull request to address issue #413 

Feel free to push back if you don't like the changes. 

* In my changes, I renamed the MANIFEST_URL variable to be MANIFEST_FILE for the block of code that handles the manifest file.
* I first check if the supplied ENV variable MANIFEST is an existing file, and use that.
* Next check is if the supplied MANIFEST is a url.  If so, use curl to download remote manifest.json file and save to /tmp, and use that file for MANIFEST_FILE.
* Last "check" is the else - to catch if it wasn't an existing file and not a URL.  I display a message but chose not to exit/fail because I didn't know if that would break anything else.  It could probably use an exit between lines 87-88.

I also fixed what I believe is a bug in the start-utils isURL function since my changes are using that function.

### CODE REVIEW

File: start-finalSetup02Modpack [hunk 1]
```diff
@@ -78,18 +78,26 @@ done
 fi
 
 if [[ "$MANIFEST" ]]; then
-EFFECTIVE_MANIFEST_URL=$(curl -Ls -o /dev/null -w %{url_effective} $MANIFEST)
-case "X$EFFECTIVE_MANIFEST_URL" in
+    if [[ -e "$MANIFEST" ]]; then
+        EFFECTIVE_MANIFEST_FILE=$MANIFEST
+    elif isURL "$MANIFEST"; then
+        EFFECTIVE_MANIFEST_FILE=/tmp/manifest.json
+        curl -Ls -o $EFFECTIVE_MANIFEST_FILE "$MANIFEST"
```
Comment (itzg) [inline start-finalSetup02Modpack]: A previous contribution (https://github.com/itzg/docker-minecraft-server/commit/df9909a8e4c499f5d6b9c0bc327a7bba6ea52a9a) discovered that the weird-looking-two-step process was needed to resolve the effective URL and then download the file from that.

I think this combination of the previous line and yours would do it:
```suggestion
        EFFECTIVE_MANIFEST_URL=$(curl -Ls -o /dev/null -w %{url_effective} $MANIFEST)
        curl -Ls -o $EFFECTIVE_MANIFEST_FILE "$EFFECTIVE_MANIFEST_URL"
```

File: start-utils [hunk 2]
```diff
@@ -3,7 +3,7 @@
 function isURL {
   local value=$1
 
-  if [[ ${value:0:8} == "https://" || ${value:0:7} = "http://" ]]; then
+  if [[ ${value:0:8} == "https://" || ${value:0:7} == "http://" ]]; then
```
Comment (itzg) [inline start-utils]: I wonder if this quietly worked since most sites are https now.

File: start-finalSetup02Modpack [hunk 3]
```diff
@@ -78,18 +78,26 @@ done
...
+    else
+        echo "MANIFEST='$MANIFEST' is not a valid manifest url or location"
```
Comment (itzg) [inline start-finalSetup02Modpack]: Yeah, I think an exit in this condition makes sense to me too.

File: start-utils (hunk 2 above)
Comment (hannah98) [reply start-utils]: I ran some tests on it and the old way worked too... I [looked it up](https://unix.stackexchange.com/a/72041/1842) and apparently both ways work - I learned something today.

Probably good to be consistent I guess.

File: start-finalSetup02Modpack (hunk 1 above)
Comment (hannah98) [reply start-finalSetup02Modpack]: Made these changes and pushed in new commit

File: start-finalSetup02Modpack (hunk 3 above)
Comment (hannah98) [reply start-finalSetup02Modpack]: Added exit in new commit

File: start-utils (hunk 2 above)
Comment (itzg) [reply start-utils]: That's good actually, because that was the same thing I was understanding from my man-page reading.

**Pull request merged**